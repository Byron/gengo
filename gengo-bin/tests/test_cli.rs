use insta::assert_snapshot;

macro_rules! assert_stdout_snapshot {
    ($cli_args:expr $(,)?) => {
        {
            let cli = gengo_bin::cli::new_from($cli_args);
            let mut buf = Vec::new();
            cli.run(&mut buf).unwrap();
            let stdout = String::from_utf8(buf).unwrap();
            assert_snapshot!(stdout);
        }
    }
}

#[test]
fn it_works() {
    assert_stdout_snapshot!(&["gengo", "1", "2"]);
}
