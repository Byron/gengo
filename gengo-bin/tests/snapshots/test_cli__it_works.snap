---
source: gengo-bin/tests/test_cli.rs
expression: stdout
---
1 + 2 = 3

